olive/platform_sdk/qualcomm/utils/data_loader.py

```diff
@@ -17,27 +16,29 @@
 logger = logging.getLogger(__name__)
 
 
-class FileListDataLoader(ABC):
-    """Abstraction for logical "FileListDataLoader", it contains data path and related metadata."""
+class FileListDataLoader:
+    """Data loader for use with Qualcomm SDKs that loads data from a file list."""
 
-    def __init__(self, config: dict, batch_size: int = None):
+    def __init__(self, load_data_func: Callable, batch_size: int = None, **kwargs):
```

Comment:
could we call the function in `__iter__` to avoid passing it by parameter to `__init__`?